File: 151003/Alekseeva/TaskKafka/Module.Publisher/Mappers/TweetMapper.cs
using Publisher.Dto.Request.CreateTo;
using Publisher.Dto.Response;
using Publisher.Models;
using Riok.Mapperly.Abstractions;
namespace Publisher.Mappers;

[Mapper]
public static partial class TweetMapper
{
    public static partial Tweet ToEntity(this CreateTweetRequestTo createTweetRequestTo);
    public static partial TweetResponseTo ToResponse(this Tweet tweet);
    public static partial IEnumerable<TweetResponseTo> ToResponse(this IEnumerable<Tweet> tweets);
}

File: 151003/Alamov/Labs-3-4/publisher/src/main/java/org/example/publisher/impl/tweet/mapper/TweetMapper.java
package org.example.publisher.impl.tweet.mapper;

import org.example.publisher.impl.editor.Editor;
import org.example.publisher.impl.tweet.Tweet;
import org.example.publisher.impl.tweet.dto.TweetRequestTo;
import org.example.publisher.impl.tweet.dto.TweetResponseTo;

import java.util.List;

public interface TweetMapper {

    TweetRequestTo tweetToRequestTo(Tweet tweet);

    List<TweetRequestTo> tweetToRequestTo(Iterable<Tweet> tweets);

    Tweet dtoToEntity(TweetRequestTo tweetRequestTo, Editor editor);

    TweetResponseTo tweetToResponseTo(Tweet tweet);

    List<TweetResponseTo> tweetToResponseTo(Iterable<Tweet> tweets);
}
